File: aspapi/Controllers/OperationController.cs
using Microsoft.AspNetCore.Mvc;

namespace aspapi.Controllers;

[ApiController]
[Route("[controller]")]
public class OperationController : ControllerBase
{
    private readonly ILogger<OperationController> _logger;
    private readonly IRepository<Operation> _repository;

    public OperationController(ILogger<OperationController> logger,IRepository<Operation> repository)
    {
        _logger = logger;
        _repository=repository;
    }

[HttpGet]
    public async Task<IEnumerable<Operation>> Get()
    {
        return await _repository.GetAllAsync();
    }

    [HttpPost]
    public async Task<Operation> Post([FromBody]Operation entity){
        entity.Id=Guid.NewGuid();
 await _repository.AddAsync(entity);

 return entity;
    }
}

File: aspapi_f/Controllers/OperationController.fs
namespace aspapi_f.Controllers

open System
open System.Collections.Generic
open System.Linq
open System.Threading.Tasks
open Microsoft.AspNetCore.Mvc
open Microsoft.Extensions.Logging
open aspapi_f

[<ApiController>]
[<Route("[controller]")>]
type OperationController(logger: ILogger<OperationController>) =
    inherit ControllerBase()

    let summaries =
        [| "Freezing"
           "Bracing"
           "Chilly"
           "Cool"
           "Mild"
           "Warm"
           "Balmy"
           "Hot"
           "Sweltering"
           "Scorching" |]

    [<HttpGet>]
    member async _.Get() =
        let rng = System.Random()

        [| for index in 0..4 ->
               { Date = DateTime.Now.AddDays(float index)
                 TemperatureC = rng.Next(-20, 55)
                 Summary = summaries.[rng.Next(summaries.Length)] } |]
